File: backend/src/order/objects/apiTypes/CheckoutQuoteResponse.scala
package delivery.order.objects.apiTypes

import delivery.order.objects.OrderPriceBreakdown
import delivery.promotion.objects.Voucher

final case class CheckoutQuoteResponse(
    canCheckout: Boolean,
    failureReason: Option[String],
    walletBalance: Double,
    originalAmount: Double,
    discountAmount: Double,
    payableAmount: Double,
    usedVoucher: Option[Voucher],
    priceBreakdown: OrderPriceBreakdown
)

File: frontend/src/objects/order/apiTypes/CheckoutQuoteResponse.ts
import type { OrderPriceBreakdown } from '@/objects/order/OrderPriceBreakdown'
import type { Voucher } from '@/objects/shared/Voucher'

export interface CheckoutQuoteResponse {
  canCheckout: boolean
  failureReason?: string | null
  walletBalance: number
  originalAmount: number
  discountAmount: number
  payableAmount: number
  usedVoucher?: Voucher | null
  priceBreakdown: OrderPriceBreakdown
}
